lib/state.go

```diff
@@ -67,8 +68,23 @@ type State struct {
 	// TODO: maybe use https://golang.org/pkg/sync/#Pool ?
 	BPool *bpool.BufferPool
 
-	Vu, Iteration int64
-	Tags          map[string]string
+	Vu                   uint64
+	iteration            int64
+	Tags                 map[string]string
+	ScenarioName         string
+	scenarioVUID         map[string]uint64
+	IncrScIter           func() int64
+	IncrScIterGlobal     func() int64
+	scIterMx             sync.RWMutex
+	scenarioVUIter       map[string]int64
+	scIter, scIterGlobal int64
+}
+
+// Init initializes some private state fields.
+func (s *State) Init() {
+	s.scenarioVUID = make(map[string]uint64)
```

Comment:
This seems to me like it can be not a map ;)

as in reality it gets set whenever you activate the VU and then we get it through using the scenarioName so it can just as well be a single uint64

Reply:
I'm not sure it can though. We set it only once if it's not defined for a particular scenario, and `js.VU` is never destroyed so these IDs can persist after being shared among scenarios. There's no need to overwrite it on each VU activation, as the VU ID should remain static throughout the test run.